game/src/core/System.java

```diff
@@ -36,4 +50,64 @@ public void run() {
     public void stop() {
         run = false;
     }
+
+    /**
+     * Use this Stream to iterate over all active entities for this system in the {@link
+     * #systemUpdate()}-Method.
+     *
+     * @return active entities that will be processed by the system as stream
+     */
+    protected Stream<Entity> getEntityStream() {
+        return entities.getSet().stream();
+    }
+
+    /**
+     * Add the given entity to the {@link DelayedSet} of entities that will be processed by the
+     * system.
+     *
+     * @see DelayedSet
+     * @param entity Entity to add
+     */
+    protected void addEntity(Entity entity) {
+        if (!entities.getSet().contains(entity))
+            if (entities.add(entity))
+                LOGGER.log(
+                        CustomLogLevel.INFO,
+                        "Entity " + entity + " will be added to the " + getClass().getName());
+    }
+
+    /**
+     * Remove the given entity to the {@link DelayedSet} of entities that will no longer be
+     * processed by the system.
+     *
+     * @see DelayedSet
+     * @param entity Entity to remove
+     */
+    protected void removeEntity(Entity entity) {
+        if (entities.getSet().contains(entity)) {
+            entities.remove(entity);
```

Comment:
Kriegen wir damit ein Problem?

`entities.remove(entity)` packt die Entität in die interne toRemove-Menge. Beim nächsten `entities.update()` wird diese Entität (a) aus der internen `current`-Menge entfernt und (b) werden ihr alle Components weggenommen. (a) dürfte korrekt sein, (b) passt hier irgendwie nicht - was ist mit anderen Systemen, die eventuell mit dieser Entität weiterarbeiten wollen?

Ich weiss noch, dass die Idee mit dem "nimm dem Ding einfach die Components weg" irgendwie schlau war. Aber ich krieg es nicht mehr zusammen, jedenfalls nicht in diesem Kontext hier.

Reply:
argl, hab ich nicht gesehen.